prov/kubernetes/execution.go

```diff
@@ -520,11 +516,9 @@ func (e *execution) deployNode(ctx context.Context, clientset kubernetes.Interfa
 		}
 		var s string
 		for _, val := range serv.Spec.Ports {
-			kubConf := e.cfg.Infrastructures["kubernetes"]
-			kubMasterIP := kubConf.GetString("master_url")
-			u, _ := url.Parse(kubMasterIP)
-			h := strings.Split(u.Host, ":")
-			str := fmt.Sprintf("http://%s:%d", h[0], val.NodePort)
+			node, _ := getHealthyNode(clientset)
```

Comment:
why is this done in the for loop? while in the previous use above it is done outside of the loop?
This should reduce API usage.